core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -496,8 +496,25 @@ public static int getSecondaryColumnIndex(int index) {
         return getPrimaryColumnIndex(index) + 1;
     }
 
-    public static long getTimestampIndexValue(long timestampIndex, long indexRow) {
-        return Unsafe.getUnsafe().getLong(timestampIndex + indexRow * 16);
+    public static long getTimestampIndexValue(long timestampIndexAddr, long indexRow) {
+        return Unsafe.getUnsafe().getLong(timestampIndexAddr + indexRow * 16);
+    }
+
+    public static void validateDesignatedTimestampBounds(long timestamp) {
+        if (timestamp == Numbers.LONG_NULL) {
+            throw CairoException.nonCritical().put("designated timestamp column cannot be NULL");
+        }
+        if (timestamp < Timestamps.O3_MIN_TS) {
+            throw CairoException.nonCritical().put("designated timestamp before 1970-01-01 is not allowed");
+        }
+    }
+
+    public static void validatePartitioningTimestampBounds(long timestamp) {
+        validateDesignatedTimestampBounds(timestamp);
```

Comment:
i don't get why do we have two separate validation methods that are no used consistently?

Reply:
Partitioning timestamp has more restrictions than a general designated timestamp. A non-partitioned table can have a designated timestamp, and we currently don't enforce the upper bound in that case.

Reply:
I changed the code to apply the same restriction in a non-partitioned table as well.